tests/cross/src/test/scala/tests/pc/InlineValueSuite.scala

```diff
@@ -213,6 +213,106 @@ class InlineValueSuite extends BaseCodeActionSuite with CommonMtagsEnrichments {
     InlineErrors.notLocal,
   )
 
+  // --- scoping tests ----------
+  checkEdit(
+    "scoping",
+    """|object Demo {
+       |  val j: Int = 5
+       |  val f: Int = 4 - j
+       |
+       |  def m() = {
+       |    val j = 10
+       |    val z = <<f>> + 1
+       |  }
+       |}""".stripMargin,
+    """|object Demo {
+       |  val j: Int = 5
+       |  val f: Int = 4 - j
+       |
+       |  def m() = {
+       |    val j = 10
+       |    val z = (4 - Demo.j) + 1
+       |  }
+       |}""".stripMargin,
+  )
+
+  checkEdit(
+    "scoping-class",
+    """|class Demo {
+       |  val j: Int = 5
+       |  val f: Int = 4 - j
+       |
+       |  def m() = {
+       |    val j = 10
+       |    val z = <<f>> + 1
+       |  }
+       |}""".stripMargin,
+    """|class Demo {
+       |  val j: Int = 5
+       |  val f: Int = 4 - j
+       |
+       |  def m() = {
+       |    val j = 10
+       |    val z = (4 - this.j) + 1
+       |  }
+       |}""".stripMargin,
+  )
+
+  // This can be improved by checking if the scope is changed
+  // an allowing to inline with impilicits otherwise.
+  // Alternatively, we can always allow but still display a warning.
+  checkErorr(
+    "scoping-implicit",
+    """|object Demo {
+       |  implicit val b : Boolean = true
+       |  def myF(implicit b : Boolean): Int = if(b) 0 else 1
+       |  val f: Int = myF
+       |
+       |  def m() = {
+       |    val z = <<f>>
+       |  }
+       |}""".stripMargin,
+    InlineErrors.impilictUsedInDefinition(3),
+  )
+
+  // This can be improved by creating a copy of the value before entering the scope
+  // object Demo {
+  //  def oo(j : Int) = {
+  //    def j_copy = j
+  //    def kk() = {
+  //      val j = 0
+  //      j_copy + 3
+  //    }
+  //  }
+  // }
+  checkErorr(
+    "bad-scoping",
+    """|object Demo {
+       |  def oo(j : Int) = {
+       |    val m = j + 3
+       |    def kk() = {
+       |      val j = 0
+       |      <<m>>
+       |    }
+       |  }
+       |}""".stripMargin,
+    InlineErrors.cannotReachVariable("Demo.j"),
```

Comment:
What about those improvements?

Reply:
I am torn about this one. On one hand this is a rare scenario and users might even be ok with inlining in this case, but on the other hand it might break the code :thinking: